x/cosmwasmpool/store.go

```diff
@@ -12,9 +14,15 @@ import (
 )
 
 // SetPool stores the given pool in state.
-func (k Keeper) SetPool(ctx sdk.Context, pool types.CosmWasmExtension) {
+func (k Keeper) SetPool(ctx sdk.Context, pool types.CosmWasmExtension) error {
+	poolModel, ok := pool.(*model.Pool)
+	if !ok {
+		return errors.New("invalid pool type when setting cosmwasm pool")
+	}
 	store := ctx.KVStore(k.storeKey)
-	osmoutils.MustSet(store, types.FormatPoolsPrefix(pool.GetId()), pool.GetStoreModel())
+	key := types.FormatPoolsPrefix(pool.GetId())
+	osmoutils.MustSet(store, key, poolModel)
+	return nil
```

Comment:
Why do we need these changes? Old logic with `pool.GetStoreModel()` looks correct to me at first glance.

The store model is what needs to be serialized to state, not the wrapper with `wasmKeeper`.

Also, by casting the input to `model.Pool`, this essentially becomes a store model with `wasmKeeper` stripped away

Reply:
This is an unnecessary change, was just done in the debugging process. Reverted https://github.com/osmosis-labs/osmosis/pull/5696/commits/a7590c0e1a8523d2d582528b1e31982d00362f12